src/test/scala/sigmastate/helpers/SigmaTestingCommons.scala

```diff
@@ -169,7 +169,10 @@ trait SigmaTestingCommons extends PropSpec
     // using default sigma reader/writer
     val bytes = serializer.toBytes(v)
     bytes.nonEmpty shouldBe true
-    serializer.parse(SigmaSerializer.startReader(bytes)) shouldBe v
+    val r = SigmaSerializer.startReader(bytes)
+    val positionLimitBefore = r.positionLimit
+    serializer.parse(r) shouldBe v
+    r.positionLimit shouldBe positionLimitBefore
```

Comment:
@greenhat this positionLimit routine shouldn't leak out of serialiser. If the serialiser need to change it internally, then it should restore the original value. This should be the contract of valid serialiser.

Reply:
My thoughts exactly. Here I just left a safeguard in test in case some serializer forgets to restore it. It's a part of our test suite, so I don't see a problem exposing some internals here.

Reply:
ok